### PR TITLE
Clean up and finalize switching RCs to Deployments in performance tests

### PR DESCRIPTION
**Pull request by mm4tt**
For the justification behind the new threshold and in general going forward with Deployments instead of RCs see https://github.com/kubernetes/kubernetes/issues/79898#issuecomment-511393121

### CODE REVIEW

File: clusterloader2/testing/density/config.yaml [hunk 1]
```diff
@@ -136,15 +136,15 @@ steps:
     replicasPerNamespace: {{$latencyReplicas}}
     tuningSet: Uniform5qps
     objectBundle:
-    - basename: latency-pod-rc
+    - basename: latency-pod-deployment
```
Comment (wojtek-t) [inline clusterloader2/testing/density/config.yaml]: latency-deployment?

File: clusterloader2/testing/density/config.yaml [hunk 2]
```diff
@@ -157,10 +157,10 @@ steps:
     replicasPerNamespace: 0
     tuningSet: Uniform5qps
     objectBundle:
-    - basename: latency-pod-rc
+    - basename: latency-pod-deployment
```
Comment (wojtek-t) [inline clusterloader2/testing/density/config.yaml]: latency-deployment?

Comment (mm4tt) [reply clusterloader2/testing/density/config.yaml]: Done.

File: clusterloader2/testing/density/config.yaml (hunk 1 above)
Comment (mm4tt) [reply clusterloader2/testing/density/config.yaml]: Done.

**Comment by wojtek-t** [comment]
/lgtm
/approve

/hold
@mm4tt - let's merge it tomorrow morning, so that we don't break anything unintentially.

**Pull request merged**